### PR TITLE
Fixes the error where DataResult is called multiple times

### PR DESCRIPTION
**Pull request by dhmaciel**
We have a crash registered on our Crashlytics related to an issue in the Pay library. It's calling the result more than once, resulting in a crash from the lib.
Testing on the library, I was able to reproduce it:
image

The solution was to prevent calls for dataResult callback more than one time per instance.

To fix it I created a flag to prevent the dataResult from being called multiple times.

### CODE REVIEW

File: pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt [hunk 1]
```diff
@@ -95,13 +96,14 @@ class GooglePayHandler(private val activity: Activity) :
...
-                activity,
-                Wallet.WalletOptions.Builder()
-                        .setEnvironment(environmentConstant)
-                        .build())
+            activity,
+            Wallet.WalletOptions.Builder()
+                .setEnvironment(environmentConstant)
+                .build()
+        )
     }
```
Comment (ViniciusDeep) [inline pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt:107]:
```
private fun paymentClientForProfile(paymentProfile: JSONObject): PaymentsClient {
    val environment = paymentProfile.optString("environment")
    val environmentConstant = PaymentsUtil.environmentForString(environment)

    val walletOptions = Wallet.WalletOptions.Builder()
        .setEnvironment(environmentConstant)
        .build()

    return Wallet.getPaymentsClient(activity, walletOptions)
}
```

Reduced Casting: Directly using optString eliminates the need for casting, making the code cleaner and less error-prone.

File: pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt [hunk 2]
```diff
@@ -228,12 +234,16 @@ class GooglePayHandler(private val activity: Activity) :
...
      * @see [
      * Wallet constants library](https://developers.google.com/android/reference/com/google/android/gms/wallet/WalletConstants.constant-summary)
      */
-    private fun handleError(statusCode: Int) =
-            loadPaymentDataResult.error(statusCode.toString(), "", null)
+    private fun handleError(statusCodeStr: String, errorMessage: String = "") {
+        if (isResponseSent) return@handleError
+        loadPaymentDataResult.error(statusCodeStr.toString(), errorMessage, null)
+        isResponseSent = true
+    }
```
Comment (ViniciusDeep) [inline pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt:248]:
```suggestion
    private fun handleError(statusCode: String, errorMessage: String = "") {
    if (isResponseSent) return

    hasResponseBeenSent = true
    loadPaymentDataResult.error(statusCode, errorMessage, null)
}

```

We can simplify the return statement: removing the unnecessary @handleError label in the return statement, as it’s redundant here.

**Review by ViniciusDeep — COMMENTED** [review]
Some of changes are fomarting maybe because the configuration into your IDE, maybe we can roll it back and just provide the necessary fix

File: pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt [hunk 3]
```diff
@@ -41,9 +41,10 @@ private const val LOAD_PAYMENT_DATA_REQUEST_CODE = 991
  * @property activity the activity used by the plugin binding.
  */
 class GooglePayHandler(private val activity: Activity) :
-        PluginRegistry.ActivityResultListener {
+    PluginRegistry.ActivityResultListener {
 
     private lateinit var loadPaymentDataResult: Result
+    private var isResponseSent = false
```
[the remark this answers is not in the document]
Comment (dhmaciel) [reply pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt]: Done!

File: pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt (hunk 1 above)
Comment (dhmaciel) [reply pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt]: I'm focused only on changing necessary pieces of code related to the bug.

**Review by ViniciusDeep — COMMENTED** [review]
LGTM!

**Comment by JlUgia** [comment]
#282 fixes this issue too, so we don't need to include additional tracking. Feel free to take a look and provide feedback.
Thank you for your contribution.

**Pull request closed without merging**